src/python/pants/engine/build_files.py

```diff
@@ -243,13 +243,23 @@ def addresses_from_address_families(address_families, spec):
   if not address_families:
     raise ResolveError('Path "{}" contains no BUILD files.'.format(spec.directory))
 
+  def exclude_address(address):
```

Comment:
I feel like the exclusions should be done in `parse_address_family` rather than here. It already has the address_mapper, and addresses_from_address_families is not the only place that plucks addresses from address families. There's also `resolve_unhydrated_struct`.